### ISSUE TITLE
Keyboard navigation should ideally be more elegant

### ISSUE THREAD

**Issue by tjsavage** [PolymerElements/iron-list#115]
Currently keyboard navigation through items in iron-list gets funky, as you can only <Tab> through items that have been stamped.

I think the ideal keyboard navigation might be the following:
- Developers using iron-list would be able to set up tabindex on nodes in their template. 
- When a user <Tab>s focus to the iron-list, the first node according to tabindex of the first item in iron-list should be focused.
- Successive <Tab> presses should cycle through nodes in the template according to the developer's defined tabindex.
- Up and down arrow keys should be used to shift focus up and down items in the list. When moving focus to a new item with arrow key, the corresponding node in the template that previously had focus in the old item should receive focus in the new item. e.g. if a button in item 4 has focus, and the user hits <Tab>, the corresponding button in item 5 now has focus.
- The focused item should always be in view. If a user uses Arrow Key to focus an item that is not in view, the list should be scrolled such that it is brought into view. This scroll should be as if the list actually was scrolled - the user should be able to use arrow keys to move through every item possible in the list, not just those that are currently stamped.
- <Tab>ing out of the last node in the currently focused item should move focus away from the list. I.e. <Tab> only moves focus through a single item in the list, and then the list loses focus.
- The list should keep state of the last node in the last item that had received focused. When the user <Tab>s back to the list, this previously focused node should be the node that regains focus. This may mean scrolling the list such that this previously focused node is back in view.

This may be unwieldy and overly complicated to implement. Let's discuss further here.

**Comment by blasten** [comment]
Fixed and released in v1.2.0
